ci/dcos.sc

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env amm
+
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+
+import scala.io.Source
+
+import $file.utils
+import utils.SemVer
+import utils.SemVerRead
+
+
+/**
+ * Main function for building DCOS registry file that works
+ * with DCOS 1.12+.
+ *
+ * @param version updated in the DCOS package files
+ */
+@main
+def buildRegistry(version: SemVer) = {
+  val universePath = createCatalogPackage(version)
+  val packagingTool = getPackagingTool()
+  buildDCOS(packagingTool, universePath, version)
+}
+
+/**
+ * Creates catalog package for Marathon with version info
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param version updated in the DCOS package files
+ * @param universePath location to build out the dcos universe files
+ */
+def createCatalogPackage(
+  version: SemVer,
+  universePath: Path = pwd / 'target / 'dcos
+): Path = {
+  rm! universePath
+  mkdir! universePath
+
+  // dcos catalog template directory
+  val templateDir: Path = pwd / 'dcos
+
+  // list all files in temp
+  templateDir.toIO.listFiles.foreach(file => {
+    val catalogFile = universePath / file.getName
+
+    Source.fromFile(file).getLines.map {
+      x => if(x.contains("__FILL_IN_VERSION_HERE__")) x.replace("__FILL_IN_VERSION_HERE__", version.toReleaseString) else x }
+      .foreach(x => write.append(catalogFile, x + System.lineSeparator()))
+    }
+  )
+  universePath
+}
+
+/**
+ * Pulls the registry build tool for the specific OS.
+ * Uses the marathon version which contains the "--use-local"
+ *
+ */
+def getPackagingTool(): Path = {
+  val os = %%("uname", "-s").out.string.trim.toLowerCase
+  val packageFile = s"dcos-registry-$os"
+  val packagePath = pwd / packageFile
+  if( exists! packagePath) { println("using cached packing tool")}
+  else {
+    val download = s"https://downloads.mesosphere.io/marathon/package-registry/binaries/cli/darwin/x86-64/$packageFile"
+    %("curl", "-O", download)
+    %("chmod", "+x", packagePath)
+  }
+  return packagePath
+}
+
+/**
+ * Creates DCOS package for Marathon
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param packagingTool the registry build tool
+ * @param version used to identify the correct build file
+ */
+def buildDCOS(
```

Comment:
Technically it's not building `DC/OS` 😉.

Reply:
definitely